google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriptionAdminClient.java

```diff
@@ -360,6 +361,21 @@ public final Subscription createSubscription(Subscription request) {
     return createSubscriptionCallable().call(request);
   }
 
+  public final Subscription createSubscription(
```

Comment:
Documentation? For this and any of the other public methods.

Reply:
So this method is only added back for binary backward-compatibility. The functionality of this method is already covered by `createSubscription(TopicName topic)`, as the old `ProjectTopicName` is the subclass of the new `TopicName`.

We are trying to remove the per-pattern resource names (`ProjectTopicName` in this case) across all cloud APIs because as we scale it's harder and harder for the generator to come up with meaningful class names. Because PubSub doesn't want a major version bump so we injected code back to the generated clients as a postprocessing step.

So far for such cases we've been only adding essentials (code) back but leaving examples and documentation for all APIs ([example](https://github.com/googleapis/java-tasks/blob/7b8bc28cd8eee86c388b65df569653d337eac548/google-cloud-tasks/src/main/java/com/google/cloud/tasks/v2/CloudTasksClient.java#L1187-L1193)). It will be doable to inject the documentation and comments back but will be adding lots of pain for future maintenance. Since I think it'll be PubSub team to continue to maintain the library I want to confirm that you want me to add it.

Reply:
We should at least have a deprecated notice on this method, indicating the preference for the other createSubscription method and that this one will be removed in the next major version release. I think we do still have to document the public interface as whether or not it is being written manually or not, we need to ensure things are properly documented for our customers.

Reply:
Done. Both deprecation information and documentation is added.